Adding a reflection configuration file fails when building native using Docker

We are currently trying to update an old 0.21.2 Quarkus Project to 0.28.1. In this project we used a reflect-config.json to keep classes from being deleted. With 0.21.2 we could successfully build a native image and a docker container, but now we can't.

We already changed to quarkus.native.additional-build-args in the application.properties file with these parameters: 
```
quarkus.native.additional-build-args=\
-H:ResourceConfigurationFiles=resources/main/resource-config.json,\
-H:ReflectionConfigurationFiles=resources/main/reflect-config.json

quarkus.native.enable-jni=true
quarkus.native.enable-http-url-handler=true
quarkus.native.enable-https-url-handler=true
quarkus.native.enable-all-security-services=true
```

Trying to execute the build ends in an error that he can't find project/resources/main/reflect-config.json.

What Quarkus tries to execute is this: 
`docker run -v /home/bfr/git/minical-multi/minical/build/minical-0.2.0-native-image-source-jar:/project:z --user 1000:1001 --rm quay.io/quarkus/ubi-quarkus-native-image:19.2.1 -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=1 -H:ResourceConfigurationFiles=resources/main/resource-config.json -H:ReflectionConfigurationFiles=resources/main/reflect-config.json --initialize-at-build-time= -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy$BySpaceAndTime -jar minical-0.2.0-runner.jar -J-Djava.util.concurrent.ForkJoinPool.common.parallelism=1 -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:-AddAllCharsets -H:EnableURLProtocols=http,https --enable-all-security-services -H:-SpawnIsolates -H:+JNI --no-server -H:-UseServiceLoaderFeature -H:+StackTrace minical-0.2.0-runner`

In the earlier versions of quarkus just the build directory was bound to docker as a volume and all files were found correctly, but now it is only the native-image-source-jar and we can't figure out a way to get the reflect-config.json into it, because always on build the directory is new generated and deletes the already copied file. 

The reflect-config.json itself is in the .jar file and can't also be accessed. Is there a way to do it, like change the volume path for the `docker run` or any other way we oversaw?

We are using gradle to build our project.

Regarding that, the ResourceConfigurationFiles are not the Problem. Only ReflectionConfigurationFiles. Also tried your solution but does not work:
Error: Invalid Path entry ../resources/main/reflect-config.json
Caused by: java.nio.file.NoSuchFileException: /resources/main/reflect-config.json

Not sure if it is helpful, gradle does not really show much information in the stacktrace.

```
Initialized native services in: /home/bfr/.gradle/native
The client will now receive all logging from the daemon (pid: 6630). The daemon log file: /home/bfr/.gradle/daemon/5.6.2/daemon-6630.out.log
Starting 2nd build in daemon [uptime: 3 mins 57.937 secs, performance: 96%]
Using 8 worker leases.
Starting Build
Settings evaluated using settings file '/home/bfr/git/minical-multi/minical/settings.gradle'.
Using local directory build cache for the root build (location = /home/bfr/.gradle/caches/build-cache-1, removeUnusedEntriesAfter = 7 days).
Projects loaded. Root project using build file '/home/bfr/git/minical-multi/minical/build.gradle'.
Included projects: [root project 'minical']

> Configure project :
Evaluating root project 'minical' using build file '/home/bfr/git/minical-multi/minical/build.gradle'.
docker.name=registry.dmz-telekom-poc/registry/minical:0.2.0
All projects evaluated.
Selected primary task 'buildNative' from project :
Tasks to be executed: [task ':dockerfileZip', task ':compileJava', task ':processResources', task ':classes', task ':jar', task ':quarkusBuild', task ':assemble', task ':downloadGraalTooling', task ':extractGraalTooling', task ':installGraalVm', task ':buildNative']
:dockerfileZip (Thread[Execution worker for ':',5,main]) started.

> Task :dockerfileZip UP-TO-DATE
Caching disabled for task ':dockerfileZip' because:
  Caching has not been enabled for the task
Skipping task ':dockerfileZip' as it is up-to-date.
:dockerfileZip (Thread[Execution worker for ':',5,main]) completed. Took 0.006 secs.
:compileJava (Thread[Execution worker for ':',5,main]) started.

> Task :compileJava UP-TO-DATE
Caching disabled for task ':compileJava' because:
  Gradle does not know how file 'build/classes/java/main/application.properties' was created (output property 'destinationDir'). Task output caching requires exclusive access to output paths to guarantee correctness.
Skipping task ':compileJava' as it is up-to-date.
:compileJava (Thread[Execution worker for ':',5,main]) completed. Took 0.016 secs.
:processResources (Thread[Execution worker for ':',5,main]) started.

> Task :processResources UP-TO-DATE
Caching disabled for task ':processResources' because:
  Caching has not been enabled for the task
Skipping task ':processResources' as it is up-to-date.
:processResources (Thread[Execution worker for ':',5,main]) completed. Took 0.002 secs.
:classes (Thread[Execution worker for ':',5,main]) started.

> Task :classes UP-TO-DATE
Skipping task ':classes' as it has no actions.
:classes (Thread[Execution worker for ':',5,main]) completed. Took 0.0 secs.
:jar (Thread[Execution worker for ':',5,main]) started.

> Task :jar
Caching disabled for task ':jar' because:
  Caching has not been enabled for the task
Task ':jar' is not up-to-date because:
  Input property 'rootSpec$1' file /home/bfr/git/minical-multi/minical/build/classes/java/main/application.properties has changed.
:jar (Thread[Execution worker for ':',5,main]) completed. Took 0.014 secs.
:quarkusBuild (Thread[Execution worker for ':',5,main]) started.
Building thin jar: /home/bfr/git/minical-multi/minical/build/minical-0.2.0-runner.jar

> Task :quarkusBuild
Caching disabled for task ':quarkusBuild' because:
  Caching has not been enabled for the task
Task ':quarkusBuild' is not up-to-date because:
  Task has not declared any outputs despite executing actions.
building quarkus runner
These dependencies are not recommended:
        - javax.activation:activation should be replaced by com.sun.activation:jakarta.activation
        - javax.enterprise:cdi-api should be replaced by jakarta.enterprise:jakarta.enterprise.cdi-api
        - javax.inject:javax.inject should be replaced by jakarta.inject:jakarta.inject-api
You might end up with two different versions of the same classes or with an artifact you shouldn't have in your classpath.
Beginning quarkus augmentation
Quarkus augmentation completed in 621ms
:quarkusBuild (Thread[Execution worker for ':',5,main]) completed. Took 0.81 secs.
:assemble (Thread[Execution worker for ':',5,main]) started.

> Task :assemble
Skipping task ':assemble' as it has no actions.
:assemble (Thread[Execution worker for ':',5,main]) completed. Took 0.0 secs.
:downloadGraalTooling (Thread[Execution worker for ':',5,main]) started.

> Task :downloadGraalTooling SKIPPED
Skipping task ':downloadGraalTooling' as task onlyIf is false.
:downloadGraalTooling (Thread[Execution worker for ':',5,main]) completed. Took 0.0 secs.
:extractGraalTooling (Thread[Execution worker for ':',5,main]) started.

> Task :extractGraalTooling SKIPPED
Skipping task ':extractGraalTooling' as task onlyIf is false.
:extractGraalTooling (Thread[Execution worker for ':',5,main]) completed. Took 0.0 secs.
:installGraalVm (Thread[Execution worker for ':',5,main]) started.

> Task :installGraalVm
Caching disabled for task ':installGraalVm' because:
  Caching has not been enabled for the task
Task ':installGraalVm' is not up-to-date because:
  Task has not declared any outputs despite executing actions.
Starting process 'command '/home/bfr/.gradle/caches/com.palantir.graal/19.2.1/graalvm-ce-19.2.1/bin/gu''. Working directory: /home/bfr/git/minical-multi/minical Command: /home/bfr/.gradle/caches/com.palantir.graal/19.2.1/graalvm-ce-19.2.1/bin/gu install native-image
Successfully started process 'command '/home/bfr/.gradle/caches/com.palantir.graal/19.2.1/graalvm-ce-19.2.1/bin/gu''
Downloading: Component catalog from www.graalvm.org
Processing component archive: Native Image
Component Native Image (org.graalvm.native-image) is already installed.
:installGraalVm (Thread[Execution worker for ':',5,main]) completed. Took 1.283 secs.
:buildNative (Thread[Execution worker for ':',5,main]) started.
Building native image source jar: /home/bfr/git/minical-multi/minical/build/minical-0.2.0-native-image-source-jar/minical-0.2.0-runner.jar
Building native image from /home/bfr/git/minical-multi/minical/build/minical-0.2.0-native-image-source-jar/minical-0.2.0-runner.jar
Running Quarkus native-image plugin on OpenJDK 64-Bit Server VM
docker run -v /home/bfr/git/minical-multi/minical/build/minical-0.2.0-native-image-source-jar:/project:z --user 1000:1001 -v $(pwd)/minical/build/:/project/build:z --rm quay.io/quarkus/ubi-quarkus-native-image:19.2.1 -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=1 -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -H:ResourceConfigurationFiles=build/resources/main/resource-config.json -H:ReflectionConfigurationFiles=build/resources/main/reflect-config.json --initialize-at-build-time= -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy$BySpaceAndTime -jar minical-0.2.0-runner.jar -J-Djava.util.concurrent.ForkJoinPool.common.parallelism=1 -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:-AddAllCharsets -H:EnableURLProtocols=http,https --enable-all-security-services -H:-SpawnIsolates -H:+JNI --no-server -H:-UseServiceLoaderFeature -H:+StackTrace minical-0.2.0-runner
docker: Error response from daemon: create $(pwd)/minical/build/: "$(pwd)/minical/build/" includes invalid characters for a local volume name, only "[a-zA-Z0-9][a-zA-Z0-9_.-]" are allowed. If you intended to pass a host directory, use absolute path.
See 'docker run --help'.

> Task :buildNative FAILED
Custom actions are attached to task ':buildNative'.
Caching disabled for task ':buildNative' because:
  Caching has not been enabled for the task
Task ':buildNative' is not up-to-date because:
  Task has not declared any outputs despite executing actions.
/home/bfr/.gradle/caches/com.palantir.graal/19.2.1/graalvm-ce-19.2.1
building native image
These dependencies are not recommended:
        - javax.activation:activation should be replaced by com.sun.activation:jakarta.activation
        - javax.enterprise:cdi-api should be replaced by jakarta.enterprise:jakarta.enterprise.cdi-api
        - javax.inject:javax.inject should be replaced by jakarta.inject:jakarta.inject-api
You might end up with two different versions of the same classes or with an artifact you shouldn't have in your classpath.
Beginning quarkus augmentation
:buildNative (Thread[Execution worker for ':',5,main]) completed. Took 0.724 secs.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':buildNative'.
> Failed to generate a native image

* Try:
Run with --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':buildNative'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$3.accept(ExecuteActionsTaskExecuter.java:166)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$3.accept(ExecuteActionsTaskExecuter.java:163)
        at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:191)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:156)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:108)
        at org.gradle.api.internal.tasks.execution.ResolveBeforeExecutionOutputsTaskExecuter.execute(ResolveBeforeExecutionOutputsTaskExecuter.java:67)
        at org.gradle.api.internal.tasks.execution.StartSnapshotTaskInputsBuildOperationTaskExecuter.execute(StartSnapshotTaskInputsBuildOperationTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.ResolveAfterPreviousExecutionStateTaskExecuter.execute(ResolveAfterPreviousExecutionStateTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:94)
        at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:95)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:406)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:102)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
        at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:43)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:355)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:343)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:336)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:322)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker$1.execute(DefaultPlanExecutor.java:134)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker$1.execute(DefaultPlanExecutor.java:129)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:202)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:193)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:129)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
Caused by: org.gradle.api.GradleException: Failed to generate a native image
        at io.quarkus.gradle.tasks.QuarkusNative.buildNative(QuarkusNative.java:421)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:103)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:49)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:42)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:28)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:717)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:684)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$5.run(ExecuteActionsTaskExecuter.java:476)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:461)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:444)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.access$200(ExecuteActionsTaskExecuter.java:93)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.execute(ExecuteActionsTaskExecuter.java:237)
        at org.gradle.internal.execution.steps.ExecuteStep.lambda$execute$1(ExecuteStep.java:33)
        at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:33)
        at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:26)
        at org.gradle.internal.execution.steps.CleanupOutputsStep.execute(CleanupOutputsStep.java:58)
        at org.gradle.internal.execution.steps.CleanupOutputsStep.execute(CleanupOutputsStep.java:35)
        at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:48)
        at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:33)
        at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:39)
        at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:73)
        at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:54)
        at org.gradle.internal.execution.steps.CatchExceptionStep.execute(CatchExceptionStep.java:35)
        at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:51)
        at org.gradle.internal.execution.steps.SnapshotOutputsStep.execute(SnapshotOutputsStep.java:45)
        at org.gradle.internal.execution.steps.SnapshotOutputsStep.execute(SnapshotOutputsStep.java:31)
        at org.gradle.internal.execution.steps.CacheStep.executeWithoutCache(CacheStep.java:208)
        at org.gradle.internal.execution.steps.CacheStep.execute(CacheStep.java:70)
        at org.gradle.internal.execution.steps.CacheStep.execute(CacheStep.java:45)
        at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:49)
        at org.gradle.internal.execution.steps.StoreSnapshotsStep.execute(StoreSnapshotsStep.java:43)
        at org.gradle.internal.execution.steps.StoreSnapshotsStep.execute(StoreSnapshotsStep.java:32)
        at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:38)
        at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:24)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:96)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$0(SkipUpToDateStep.java:89)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:54)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:38)
        at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:76)
        at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:37)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:36)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:26)
        at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:90)
        at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:48)
        at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:69)
        at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:47)
        at org.gradle.internal.execution.impl.DefaultWorkExecutor.execute(DefaultWorkExecutor.java:33)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:140)
        ... 35 more
Caused by: io.quarkus.creator.AppCreatorException: Failed to augment application classes
        at io.quarkus.creator.phase.augment.AugmentTask.run(AugmentTask.java:190)
        at io.quarkus.creator.phase.augment.AugmentTask.run(AugmentTask.java:48)
        at io.quarkus.creator.CuratedApplicationCreator.runTask(CuratedApplicationCreator.java:139)
        at io.quarkus.gradle.tasks.QuarkusNative.buildNative(QuarkusNative.java:419)
        ... 89 more
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.deployment.pkg.steps.NativeImageBuildStep#build threw an exception: java.lang.RuntimeException: Failed to build native image
        at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:294)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at io.quarkus.deployment.ExtensionLoader$1.execute(ExtensionLoader.java:941)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:415)
        at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
        at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1535)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
        at java.lang.Thread.run(Thread.java:748)
        at org.jboss.threads.JBossThread.run(JBossThread.java:479)
Caused by: java.lang.RuntimeException: Image generation failed
        at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:283)
        ... 12 more

        at io.quarkus.builder.Execution.run(Execution.java:108)
        at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:121)
        at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:130)
        at io.quarkus.creator.phase.augment.AugmentTask.run(AugmentTask.java:181)
        ... 92 more
Caused by: java.lang.RuntimeException: Failed to build native image
        at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:294)
        at io.quarkus.deployment.ExtensionLoader$1.execute(ExtensionLoader.java:941)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:415)
        at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
        at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1535)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
        at org.jboss.threads.JBossThread.run(JBossThread.java:479)
Caused by: java.lang.RuntimeException: Image generation failed
        at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:283)
        ... 7 more


* Get more help at https://help.gradle.org

BUILD FAILED in 4s
7 actionable tasks: 4 executed, 3 up-to-date

```

Do I have to do anything specific to reproduce the error with your sample?

I am getting:

```
> Configure project :
docker.name=example.com/registry/bug-example:0.2.0
ERROR: Local folder /home/gandrian/Desktop/quarkus-examples/reflection-bug-example/src/test/resources doesn't exist. Mounted volume will not be visible to container

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'reflection-bug-example'.
> Local folder /home/gandrian/Desktop/quarkus-examples/reflection-bug-example/src/test/resources doesn't exist.
```

when doing `./gradlew buildNative`

I am seeing:

```
> Task :installGraalVm
Downloading: Component catalog from www.graalvm.org
Processing component archive: Native Image
Component Native Image (org.graalvm.native-image) is already installed.
Error: Invalid Path entry build/resources/main/reflect-config.json
Caused by: java.nio.file.NoSuchFileException: /project/build/resources/main/reflect-config.json
```

Notice ` /project/build/resources/main/reflect-config.json` which obviously doesn't exist. Is your build script doing anything weird that would cause `/project` to be used at the root of the project?

I am not really sure what's wrong with the whole setup (I checked with another project and the configuration in `application.properties` does work correctly)

but when I do comment out the properties and instead add this configuration: 

```
buildNative {
    additionalBuildArgs = [
            "-H:ReflectionConfigurationFiles=${buildDir}/resources/main/reflect-config.json".toString(),
            "-H:ResourceConfigurationFiles=${buildDir}/resources/main/resource-config.json".toString()
    ]
}

then I get: 


```
> Task :installGraalVm
Downloading: Component catalog from www.graalvm.org
Processing component archive: Native Image
Component Native Image (org.graalvm.native-image) is already installed.
Error: Invalid Path entry /home/gandrian/Desktop/quarkus-examples/reflection-bug-example/build/resources/main/reflect-config.json
Caused by: java.nio.file.NoSuchFileException: /home/gandrian/Desktop/quarkus-examples/reflection-bug-example/build/resources/main/reflect-config.json
```

In this case the paths are definitely correct so I am not sure what's going on

--- a/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java
@@ -10,6 +10,7 @@ import java.io.FileOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
+import java.io.UncheckedIOException;
 import java.nio.file.FileAlreadyExistsException;
 import java.nio.file.FileSystem;
 import java.nio.file.FileVisitOption;
@@ -30,6 +31,7 @@ import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.function.BiPredicate;
 import java.util.function.BooleanSupplier;
 import java.util.function.Consumer;
 import java.util.jar.Attributes;
@@ -337,6 +339,7 @@ public class JarResultBuildStep {
                 .resolve(outputTargetBuildItem.getBaseName() + "-native-image-source-jar");
         IoUtils.recursiveDelete(thinJarDirectory);
         Files.createDirectories(thinJarDirectory);
+        copyJsonConfigFiles(applicationArchivesBuildItem, thinJarDirectory);
 
         Path runnerJar = thinJarDirectory
                 .resolve(outputTargetBuildItem.getBaseName() + packageConfig.runnerSuffix + ".jar");
@@ -356,6 +359,35 @@ public class JarResultBuildStep {
         }
         runnerJar.toFile().setReadable(true, false);
         return new NativeImageSourceJarBuildItem(runnerJar, libDir);
+    }
+
+    /**
+     * This is done in order to make application specific native image configuration files available to the native-image tool
+     * without the user needing to know any specific paths.
+     * The files that are copied don't end up in the native image unless the user specifies they are needed, all this method
+     * does is copy them to a convenient location
+     */
+    private void copyJsonConfigFiles(ApplicationArchivesBuildItem applicationArchivesBuildItem, Path thinJarDirectory)
+            throws IOException {
+        // this will contain all the resources in both maven and gradle cases - the latter is true because we copy them in AugmentTask
+        Path classesLocation = applicationArchivesBuildItem.getRootArchive().getArchiveLocation();
+        Files.find(classesLocation, 1, new BiPredicate<Path, BasicFileAttributes>() {
+            @Override
+            public boolean test(Path path, BasicFileAttributes basicFileAttributes) {
+                return basicFileAttributes.isRegularFile() && path.toString().endsWith(".json");
+            }
+        }).forEach(new Consumer<Path>() {
+            @Override
+            public void accept(Path jsonPath) {
+                try {
+                    Files.copy(jsonPath, thinJarDirectory.resolve(jsonPath.getFileName()));
+                } catch (IOException e) {
+                    throw new UncheckedIOException(
+                            "Unable to copy json config file from " + jsonPath + " to " + thinJarDirectory,
+                            e);
+                }
+            }
+        });
     }
 
     private void doThinJarGeneration(CurateOutcomeBuildItem curateOutcomeBuildItem,

--- a/docs/src/main/asciidoc/writing-native-applications-tips.adoc
+++ b/docs/src/main/asciidoc/writing-native-applications-tips.adoc
@@ -23,11 +23,11 @@ GraalVM imposes a number of constraints and making your application a native exe
 === Including resources
 
 By default, when building a native executable, GraalVM will not include any of the resources that are on the classpath into the native executable it creates.
-Such resources that are meant to be part of the native executable need to be specified explicitly.
+Resources that are meant to be part of the native executable need to be configured explicitly.
 
 Quarkus automatically includes the resources present in `META-INF/resources` (the web resources) but, outside of this directory, you are on your own.
 
-To include more resources in the native executable, create a `resources-config.json` JSON file defining which resources should be included:
+To include more resources in the native executable, create a `resources-config.json` (the most common location is within `src/main/resources`) JSON file defining which resources should be included:
 
 [source,json]
 ----
@@ -43,13 +43,41 @@ To include more resources in the native executable, create a `resources-config.j
 }
 ----
 
-When using Maven, you can put this file at the root of your project.
-If Gradle is used, we need to put this file in a folder that will be deployed in the build folder (for example `src/main/resources`) because the build folder is used as the project folder when creating the native image.
-
 The patterns are valid Java regexps.
 Here we include all the XML files and JSON files into the native executable.
 
-Now, let's tell the `native-image` executable about this configuration file by tweaking the `native-image` execution in your `pom.xml`:
+[NOTE]
+====
+You can find more information about this topic in https://github.com/oracle/graal/blob/master/substratevm/RESOURCES.md[the GraalVM documentation].
+====
+
+The final order of business is to make the configuration file known to the `native-image` executable by adding the proper configuration to `application.properties`:
+
+[source,properties]
+----
+quarkus.native.additional-build-args =-H:ResourceConfigurationFiles=resources-config.json
+----
+
+In the previous snippet we were able to simply use `resources-config.json` instead of specifying the entire path of the file simply because it was added to `src/main/resources`.
+If the file had been added to another directory, the proper file path would have had to be specified manually.
+
+[TIP]
+====
+Multiple options may be separated by a comma. For example, one could use:
+
+[source,properties]
+----
+quarkus.native.additional-build-args =\
+    -H:ResourceConfigurationFiles=resource-config.json,\
+    -H:ReflectionConfigurationFiles=reflection-config.json
+----
+
+in order to ensure that various resources are included and additional reflection is registered.
+
+====
+If for some reason adding the aforementioned configuration to `application.properties` is not desirable, it is possible to configure the build tool to effectively perform the same operation.
+
+When using Maven, we could use the following configuration:
 
 [source,xml]
 ----
@@ -58,36 +86,22 @@ Now, let's tell the `native-image` executable about this configuration file by t
         <id>native</id>
         <properties>
             <quarkus.package.type>native</quarkus.package.type>
-            <quarkus.native.additional-build-args>-H:ResourceConfigurationFiles=${project.basedir}/resources-config.json</quarkus.native.additional-build-args>
+            <quarkus.native.additional-build-args>-H:ResourceConfigurationFiles=resources-config.json</quarkus.native.additional-build-args>
         </properties>
     </profile>
 </profiles>
 ----
 
-[TIP]
-====
-Multiple options may be separated by a comma.
-
-Another possibility is to include the `quarkus.native.additional-build-args` configuration property in your `application.properties`.
-====
-
-When using Gradle, we need to configure the additional build arguments in the plugin configuration:
+When using Gradle, configuring the `buildNative` task like so suffices:
 
 [source,groovy]
 ----
 buildNative {
     additionalBuildArgs = [
-            "-H:ResourceConfigurationFiles=${buildDir}/resources/main/resources-config.json".toString()
+            '-H:ResourceConfigurationFiles=resources-config.json'
     ]
 }
 ----
-
-Your resources will now be included in your native executable.
-
-[NOTE]
-====
-You can find more information about this topic in https://github.com/oracle/graal/blob/master/substratevm/RESOURCES.md[the GraalVM documentation].
-====
 
 === Registering for reflection
 
@@ -168,7 +182,7 @@ Obviously, adding `@RegisterForReflection` is not possible if the class is in a 
 
 In this case, you can use a configuration file to register classes for reflection.
 
-Let's create the `reflection-config.json` file:
+As an example, in order to register all methods of class `com.acme.MyClass` for reflection, we create `reflection-config.json` (the most common location is within `src/main/resources`)
 
 [source,json]
 ----
@@ -185,15 +199,38 @@ Let's create the `reflection-config.json` file:
 ]
 ----
 
-When using Maven, you can put this file at the root of your project.
-If Gradle is used, we need to put this file in a folder that will be deployed in the build folder (for example `src/main/resources`) because the build folder is used as the project folder when creating the native image.
-
 [NOTE]
 ====
 For more details on the format of this file, please refer to https://github.com/oracle/graal/blob/master/substratevm/REFLECTION.md[the GraalVM documentation].
 ====
 
-Now, let's tell the `native-image` executable about this configuration file by tweaking the `native-image` execution in the `pom.xml`:
+The final order of business is to make the configuration file known to the `native-image` executable by adding the proper configuration to `application.properties`:
+
+[source,properties]
+----
+quarkus.native.additional-build-args =-H:ResourceConfigurationFiles=reflection-config.json
+----
+
+In the previous snippet we were able to simply use `reflection-config.json` instead of specifying the entire path of the file simply because it was added to `src/main/resources`.
+If the file had been added to another directory, the proper file path would have had to be specified manually.
+
+[TIP]
+====
+Multiple options may be separated by a comma. For example, one could use:
+
+[source,properties]
+----
+quarkus.native.additional-build-args =\
+    -H:ResourceConfigurationFiles=resource-config.json,\
+    -H:ReflectionConfigurationFiles=reflection-config.json
+----
+
+in order to ensure that various resources are included and additional reflection is registered.
+
+====
+If for some reason adding the aforementioned configuration to `application.properties` is not desirable, it is possible to configure the build tool to effectively perform the same operation.
+
+When using Maven, we could use the following configuration:
 
 [source,xml]
 ----
@@ -202,31 +239,22 @@ Now, let's tell the `native-image` executable about this configuration file by t
         <id>native</id>
         <properties>
             <quarkus.package.type>native</quarkus.package.type>
-            <quarkus.native.additional-build-args>-H:ReflectionConfigurationFiles=${project.basedir}/reflection-config.json</quarkus.native.additional-build-args>
+            <quarkus.native.additional-build-args>-H:ReflectionConfigurationFiles=reflection-config.json</quarkus.native.additional-build-args>
         </properties>
     </profile>
 </profiles>
 ----
 
-[TIP]
-====
-Multiple options may be separated by a comma.
-
-Another possibility is to include the `quarkus.native.additional-build-args` configuration property in your `application.properties`.
-====
-
-When using Gradle, we need to configure the additional build arguments in the plugin configuration:
+When using Gradle, configuring the `buildNative` task like so suffices:
 
 [source,groovy]
 ----
 buildNative {
     additionalBuildArgs = [
-            "-H:ReflectionConfigurationFiles=${buildDir}/resources/main/reflection-config.json".toString()
+            '-H:ReflectionConfigurationFiles=reflection-config.json'
     ]
 }
 ----
-
-`com.acme.MyClass` will now be registered for reflection and included in the native executable.
 
 === Delaying class initialization
 

